openbb_platform/providers/intrinio/openbb_intrinio/models/forward_eps_estimates.py

```diff
@@ -0,0 +1,228 @@
+    )
+    mean_1m: Optional[float] = Field(
+        default=None,
+        description="The mean estimate for the period one month ago.",
+    )
+    mean_2m: Optional[float] = Field(
+        default=None,
+        description="The mean estimate for the period two months ago.",
+    )
+    mean_3m: Optional[float] = Field(
+        default=None,
+        description="The mean estimate for the period three months ago.",
+    )
+
+    @field_validator(
+        "revisions_change_percent",
+        mode="before",
+        check_fields=False,
+    )
+    @classmethod
+    def normalize_percent(
+        cls, v: Optional[Union[int, float]]
+    ) -> Optional[Union[int, float]]:
+        """Normalize percent values."""
+        return v / 100 if v else None
+
+
+class IntrinioForwardEpsEstimatesFetcher(
+    Fetcher[
+        IntrinioForwardEpsEstimatesQueryParams, List[IntrinioForwardEpsEstimatesData]
+    ]
+):
+    """Intrinio Forward EPS Estimates Fetcher."""
+
+    @staticmethod
+    def transform_query(
+        params: Dict[str, Any]
+    ) -> IntrinioForwardEpsEstimatesQueryParams:
+        """Transform the query params."""
+        return IntrinioForwardEpsEstimatesQueryParams(**params)
+
+    @staticmethod
+    async def aextract_data(
+        query: IntrinioForwardEpsEstimatesQueryParams,
+        credentials: Optional[Dict[str, str]],
+        **kwargs: Any,
+    ) -> List[Dict]:
+        """Return the raw data from the Intrinio endpoint."""
+        api_key = credentials.get("intrinio_api_key") if credentials else ""
+
+        BASE_URL = "https://api-v2.intrinio.com/zacks/eps_estimates?page_size=10000"
+
+        symbols = query.symbol.split(",") if query.symbol else None
+
+        query_str = get_querystring(
+            query.model_dump(by_alias=True),
+            ["symbol", "calendar_period", "fiscal_period", "limit"],
+        )
+
+        results: List[Dict] = []
+
+        async def get_one(symbol):
+            """Get the data for one symbol."""
+            url = f"{BASE_URL}&identifier={symbol}&{query_str}&api_key={api_key}"
+            new_data: List[Dict] = []
+            data = await amake_request(
+                url, response_callback=response_callback, **kwargs
+            )
+            if not data or not isinstance(data, dict) or not data.get("estimates"):
+                warn(f"Symbol Error: No data found for {symbol}")
+            if isinstance(data, dict) and data.get("estimates"):
+                new_data = data.get("estimates")  # type: ignore
+                if new_data:
+                    results.extend(new_data)
+
+        if symbols:
+            await asyncio.gather(*[get_one(symbol) for symbol in symbols])
+            return results
+
+        async def fetch_callback(response, session):
+            """Callback for pagination."""
+            data = await response.json()
+            if data.get("messages"):  # type: ignore
```

Comment:
You can `get` the estimates and messages once here. No need to try grabbing them two times.